### PR TITLE
ref(mediators): Convert AlertRuleActionRequester to dataclass

### PR DESCRIPTION
**Pull request by Christinarlong**
Convert the AlertRuleActionRequester mediator to a dataclass and update refs to point to this new class. Also moves the utils for external_requests while were at it.

Would also appreciate any suggestions for how to name the directories/structure 

issue ref (#73857 )

### CODE REVIEW

File: src/sentry/sentry_apps/external_requests/alert_rule_action_requester.py
Comment (markstory) [inline src/sentry/sentry_apps/external_requests/alert_rule_action_requester.py:1]: Could we remove the mediator that this replaces too?

File: src/sentry/sentry_apps/external_requests/utils.py
Comment (markstory) [inline src/sentry/sentry_apps/external_requests/utils.py:1]: Should we update references to `sentry.mediators.external_requests.util` to point to this module? We could remove the old module that way.

File: src/sentry/sentry_apps/external_requests/alert_rule_action_requester.py [hunk 1]
```diff
@@ -25,26 +23,60 @@ class AlertRuleActionResult(TypedDict):
...
+                scope.set_tag("sentry_app", self.sentry_app.slug)
+                scope.set_context(
+                    "alert_rule_action.error",
+                    {
+                        "sentry_app_slug": self.sentry_app.slug,
+                        "install_uuid": self.install.uuid,
+                        "uri": self.uri,
+                        "error_message": str(e),
+                    },
+                )
```
Comment (markstory) [inline src/sentry/sentry_apps/external_requests/alert_rule_action_requester.py:57]: In the old implementation this was a logging message. This scope configuration doesn't capture any exceptions or messages so it won't do much. We should keep the logging message though as we don't need to be alerted on requests to sentryapps failing.

File: tests/sentry/sentry_apps/external_requests/test_alert_rule_action_requester.py [hunk 2]
```diff
@@ -1,10 +1,16 @@
...
     DEFAULT_ERROR_MESSAGE,
     DEFAULT_SUCCESS_MESSAGE,
     AlertRuleActionRequester,
 )
+
+# from sentry.mediators.external_requests.alert_rule_action_requester import (
+#     DEFAULT_ERROR_MESSAGE,
+#     DEFAULT_SUCCESS_MESSAGE,
+#     AlertRuleActionRequester,
+# )
```
Comment (markstory) [inline tests/sentry/sentry_apps/external_requests/test_alert_rule_action_requester.py:13]: Could remove this too.

Comment (Christinarlong) [reply tests/sentry/sentry_apps/external_requests/test_alert_rule_action_requester.py:13]: oops, yeah that was debugging I forgot to remove

**Review by markstory — APPROVED** [review]

**Pull request merged**